allure-testng/src/test/java/io/qameta/allure/test/ListenerTests.java

```diff
@@ -0,0 +1,212 @@
+        verify(allure).startTestContainer(suite.capture());
+        assertThat(suite.getValue())
+                .extracting(TestResultContainer::getName).contains(SUITE_NAME);
+    }
+
+    @Test
+    public void suiteFinish() {
+        ISuite iSuite = mock(ISuite.class);
+        when(iSuite.getAttribute(ALLURE_UUID_KEY)).thenReturn(SUITE_UUID);
+        allureTestNg.onFinish(iSuite);
+
+        InOrder order = inOrder(allure);
+        order.verify(allure).stopTestContainer(SUITE_UUID);
+        order.verify(allure).writeTestContainer(SUITE_UUID);
+    }
+
+    @Test
+    public void contextStart() {
+        ISuite iSuite = mock(ISuite.class);
+        when(iSuite.getAttribute(ALLURE_UUID_KEY)).thenReturn(SUITE_UUID);
+        ITestContext iContext = mock(ITestContext.class);
+        when(iContext.getAttribute(ALLURE_UUID_KEY)).thenReturn(CONTAINER_UUID);
+        when(iContext.getName()).thenReturn(CONTEXT_NAME);
+        when(iContext.getSuite()).thenReturn(iSuite);
+
+        allureTestNg.onStart(iContext);
+        ArgumentCaptor<TestResultContainer> contextContainer = ArgumentCaptor.forClass(TestResultContainer.class);
+        verify(allure).startTestContainer(eq(SUITE_UUID), contextContainer.capture());
+        assertThat(contextContainer.getValue())
+                .extracting(TestResultContainer::getUuid).contains(CONTAINER_UUID);
+        assertThat(contextContainer.getValue())
+                .extracting(TestResultContainer::getName).contains(CONTEXT_NAME);
+    }
+
+    @Test
+    public void contextFinish() {
+        ITestContext iContext = mock(ITestContext.class);
+        when(iContext.getAttribute(ALLURE_UUID_KEY)).thenReturn(CONTAINER_UUID);
+
+        allureTestNg.onFinish(iContext);
+        InOrder order = inOrder(allure);
+        order.verify(allure).stopTestContainer(CONTAINER_UUID);
+        order.verify(allure).writeTestContainer(CONTAINER_UUID);
+    }
+
+    private Method parameterizedMethod(String parameter) {
+        try {
+            return this.getClass().getDeclaredMethod("parameterizedMethod", String.class);
+        } catch (NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void mockTestStart() {
+        ITestResult iResult = mock(ITestResult.class);
+        when(iResult.getAttribute(ALLURE_UUID_KEY)).thenReturn(RESULT_UUID);
+
+        ITestContext iContext = mock(ITestContext.class);
+        when(iContext.getAttribute(ALLURE_UUID_KEY)).thenReturn(CONTAINER_UUID);
+        when(iResult.getTestContext()).thenReturn(iContext);
+
+        ITestNGMethod method = mock(ITestNGMethod.class);
+        ITestClass testClass = mock(ITestClass.class);
+        XmlTest xmlTest = mock(XmlTest.class);
+        XmlSuite suite = mock(XmlSuite.class);
+        ConstructorOrMethod constructorOrMethod = mock(ConstructorOrMethod.class);
+
+        when(iResult.getMethod()).thenReturn(method);
```

Comment:
maybe better to write real implementation?

also: mockito has option DEEP_STUBS (or smth close to it) - it can help to avoid 

```
 ITestNGMethod method = mock(ITestNGMethod.class);
 ITestClass testClass = mock(ITestClass.class);
 when(method.getTestClass()).thenReturn(testClass);
```